Tests/TuistGeneratorTests/Graph/GraphTests.swift

```diff
@@ -352,6 +352,58 @@ final class GraphTests: XCTestCase {
             "Bundle1",
         ])
     }
+
+    func test_encode() {
+        // Given
+        let cache = GraphLoaderCache()
+        let graph = Graph.test(cache: cache)
+        let framework = FrameworkNode(path: fixturePath(path: RelativePath("xpm.framework")))
+        let library = LibraryNode(path: fixturePath(path: RelativePath("libStaticLibrary.a")),
+                                  publicHeaders: fixturePath(path: RelativePath("")))
+        let target = TargetNode.test(dependencies: [framework, library])
+        cache.add(targetNode: target)
+        cache.add(precompiledNode: framework)
+        cache.add(precompiledNode: library)
+
+        let expected = """
```

Comment:
@marciniwanicki / @kwridan do you think it'd be enough for you having the `Graph` model conforming the `Encodable` protocol? The command in `TuistKit` will encode the model into the JSON format and print it to the standard output.

Reply:
I like the idea @kwridan. I'm not sure though how that API would help reduce the surface area of the graph.

Reply:
something like this will help limit the public types to this interface without needing to make the `Graphing` protocol and all the `GraphNodes` public too